go/ct/evm_test.go

```diff
@@ -102,3 +105,246 @@ func TestCt_ExplicitCases(t *testing.T) {
+				stack := [fuzzIdealStackSize * 32]byte{}
+				// fill a quarter with random values
+				var i int
+				for i = 0; i < 7*8; i++ {
+					stack[i] = byte(rnd.Int31n(256))
+				}
+				// leave a quarter with zeros
+				for ; i < 7*16; i++ {
+					stack[i] = 0
+				}
+				// fill a quarter with max values
+				for ; i < 7*24; i++ {
+					stack[i] = 255
+				}
+				// fill a quarter with avg values
+				for ; i < 7*32; i++ {
+					stack[i] = 127
+				}
+
+				f.Add(
+					ops[:],         // opCodes
+					int64(gas),     // gas
+					byte(revision), // revision
+					stack[:],       // stack
+				)
+			}
+		}
+	}
+
+	// add one more with a full stack
+	ops := [fuzzMaximumCodeSegment]byte{}
+	rnd.Read(ops[:])
+	ops[0] = byte(0x00)
+	fullStack := [1024]byte{}
+	rnd.Read(fullStack[:])
+	f.Add(
+		ops[:],             // opCodes
+		int64(0),           // gas
+		byte(R07_Istanbul), // revision
+		fullStack[:],       // stack
+	)
+}
+
+// errorReportString is a helper function to print a summary of the test states.
+// Fuzzing generates a single error at the time, but when tunning the initial test
+// data, such states are evaluated in parallel and errors are produced in large numbers.
+// This function aims to present a compact overview of the most significant differences to
+// keep process output small and readable.
+func errorReportString(original *st.State, resultState *st.State, referenceState *st.State) string {
+	return fmt.Sprintf(`
+	--- revision: %v
+	--- code(pc:%d) %v
+	--- status: testee %v, reference %v
+	--- gas: start %v, testee %v, reference %v
+	--- stack size: start %v, testee %v, reference %v
+	--- mem size: start %v, testee %v, reference %v`,
+		original.Revision,
+		original.Pc, original.Code.HumanReadableString(0, 7),
+		resultState.Status, referenceState.Status,
+		original.Gas, resultState.Gas, referenceState.Gas,
+		original.Stack.Size(), resultState.Stack.Size(), referenceState.Stack.Size(),
+		original.Memory.Size(), resultState.Memory.Size(), referenceState.Memory.Size(),
+	)
+}
+
+func FuzzDifferentialLfvmVsGeth(f *testing.F) {
+
+	gethVm := geth.NewConformanceTestingTarget()
+	lfvmVm := lfvm.NewConformanceTestingTarget()
+
+	rnd := rand.New(0)
+
+	prepareInitialTestData(f, rnd)
+
+	f.Fuzz(func(t *testing.T, opCodes []byte, gas int64, revision byte, stackBytes []byte) {
+		if gas < 0 {
+			t.Skip("negative gas", gas)
+		}
+
+		if Revision(revision) < R07_Istanbul || Revision(revision) > NewestSupportedRevision {
+			t.Skip("unsupported revision", revision)
+		}
+
+		if len(opCodes) == 0 {
+			t.Skip("empty opCodes")
+		}
+
+		if len(opCodes) > fuzzMaximumCodeSegment {
+			t.Skip("too many opCodes,  not interesting")
+		}
+
+		// Ignore stack sizes larger than 7 words, as they are not interesting
+		// Do not ignore stack sizes close to the overflow, as they are interesting
+		if len(stackBytes) > fuzzIdealStackSize*32 && len(stackBytes) < (1024-fuzzIdealStackSize) {
+			t.Skip("Uninteresting stack size", len(stackBytes))
```

Comment:
nice :+1:

Reply:
the goal here is:

- help the fuzzer to early discard mutations that do not increase coverage. 
- avoid cryptic encoding of states, as the future bugs will be stored, and they need to be potentially migrated or converted into a unit test.